pkg/oam/util/helper_test.go

```diff
@@ -292,3 +292,46 @@ var _ = Describe("Test unstructured related helper utils", func() {
 		}
 	})
 })
+
+var _ = Describe("Test GenTraitName helper utils", func() {
+	It("Test generate trait name", func() {
+
+		mts := v1alpha2.ManualScalerTrait{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "ns",
+				Name:      "sample-manualscaler-trait",
+			},
+			Spec: v1alpha2.ManualScalerTraitSpec{
+				ReplicaCount: 3,
+			},
+		}
+
+		test := []struct {
+			name     string
+			template *v1alpha2.ComponentTrait
+		}{
+			{
+				name:     "simple",
+				template: &v1alpha2.ComponentTrait{},
+			},
+		}
+		for _, test := range test {
+			expected := "simple-trait-67b8949f8d"
+			oldTraitName := util.GenTraitName(test.name, test.template)
+			Expect(oldTraitName).Should(Equal(expected))
+
+			newTemplate := test.template.DeepCopy()
+			newTemplate.Trait = runtime.RawExtension{
+				Object: &mts,
+			}
+			newTraitName := util.GenTraitName(test.name, newTemplate)
+			Expect(oldTraitName).ShouldNot(Equal(newTraitName))
+
+			newTemplate.Trait = runtime.RawExtension{}
+
+			lastedTraitName := util.GenTraitName(test.name, newTemplate)
```

Comment:
a table style test should not have logics in the last loop, instead, they should be test cases. You can move those to test cases and add a result (should equal or not equal) there.

Reply:
done